applications/MedApplication/custom_utilities/med_testing_utilities.cpp

```diff
@@ -0,0 +1,197 @@
+// KRATOS  __  __          _    _                _ _           _   _
+//        |  \/  | ___  __| |  / \   _ __  _ __ | (_) ___ __ _| |_(_) ___  _ ___
+//        | |\/| |/ _ \/ _` | / _ \ | '_ \| '_ \| | |/ __/ _` | __| |/ _ \| '_  |
+//        | |  | |  __/ (_| |/ ___ \| |_) | |_) | | | (_| (_| | |_| | (_) | | | |
+//        |_|  |_|\___|\__,_/_/   \_\ .__/| .__/|_|_|\___\__,_|\__|_|\___/|_| |_|
+//                                  |_|   |_|
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher (https://github.com/philbucher)
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "med_testing_utilities.h"
+
+
+namespace Kratos {
+
+namespace { // helpers namespace
+
+using NodeType = ModelPart::NodeType;
+using GeometryType = ModelPart::GeometryType;
+using GeometryContainerType = ModelPart::GeometryContainerType;
+
+template<class T>
+bool contains(
+    const std::vector<T> rVec,
+    const T& rValue)
+{
+    return std::find(rVec.begin(), rVec.end(), rValue) != rVec.end();
+}
+
+// check if the geometries have the same nodes
+// for this to be the case, two conditions need to be fulflled:
+// 1: geometries have same number of nodes
+// 2: nodes of both geometries are in same order and in same coords
+bool have_same_nodes(
+    const GeometryType& rGeom1,
+    const GeometryType& rGeom2)
+{
+    if (rGeom1.PointsNumber() != rGeom2.PointsNumber()) return false;
+
+    for (std::size_t i=0; i<rGeom1.PointsNumber(); ++i) {
+        if (rGeom1[i].Distance(rGeom2[i]) > 1e-12) return false;
+    }
+
+    return true;
+}
+
+void CheckEntitiesAreEqual(
+    const NodeType& rNode1,
+    const NodeType& rNode2)
+{
+    KRATOS_TRY
+
+    // KRATOS_CHECK_EQUAL(rNode1.Id(), rNode2.Id()); // the MEDApp deliberately does not care about IDs
+
+    KRATOS_CHECK_DOUBLE_EQUAL(rNode1.X(),  rNode2.X());
+    KRATOS_CHECK_DOUBLE_EQUAL(rNode1.X0(), rNode2.X0());
+
+    KRATOS_CHECK_DOUBLE_EQUAL(rNode1.Y(),  rNode2.Y());
+    KRATOS_CHECK_DOUBLE_EQUAL(rNode1.Y0(), rNode2.Y0());
+
+    KRATOS_CHECK_DOUBLE_EQUAL(rNode1.Z(),  rNode2.Z());
+    KRATOS_CHECK_DOUBLE_EQUAL(rNode1.Z0(), rNode2.Z0());
+
+    KRATOS_CATCH("")
+}
+
+void CheckGeometriesAreEqual(
+    const GeometryType& rGeom1,
+    const GeometryType& rGeom2)
+{
+    KRATOS_TRY
+
+    // KRATOS_CHECK_EQUAL(rGeom1.Id(), rGeom2.Id()); // the MEDApp deliberately does not care about IDs
+
+    KRATOS_CHECK_EQUAL(rGeom1.PointsNumber(), rGeom2.PointsNumber());
+
+    KRATOS_CHECK(GeometryType::IsSame(rGeom1, rGeom2));
```

Comment:
Save for this call, `CheckGeometriesAreEqual` is identical to `have_same_nodes`, right? If so, you could just replace the other checks with a call to `have_same_nodes`.

Reply:
`GeometryType::IsSame`  is not done in `have_same_nodes`

Maybe I misunderstand?